src/integrationtests/java/com/aws/greengrass/integrationtests/deployment/DeploymentConfigMergingTest.java

```diff
@@ -709,6 +742,144 @@ void GIVEN_kernel_running_service_WHEN_run_with_change_THEN_service_restarts() t
+                "xxxxxx.credentials.iot.us-east-1.amazonaws.com", "privKeyFilePath", "certFilePath", "caFilePath",
+                "us-east-1", "roleAliasName");
+        kernel.getContext().put(DeviceConfiguration.class, deviceConfiguration);
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel, getClass().getResource("config.yaml"));
+
+        // Mock Fss trigger at kernel launch
+        when(mqttClient.publish(any(PublishRequest.class))).thenReturn(CompletableFuture.completedFuture(0));
+        kernel.launch();
+
+        // Read deployment config from yaml file
+        Map<String, Object> deploymentConfig = ConfigPlatformResolver
+                .resolvePlatformMap(getClass().getResource("connectivityValidationConfig.yaml"));
+        int expectedMqttPort = 8080;        // mqtt.port
+        int expectedDataPlanePort = 443;    // greengrassDataPlanePort
+
+        // Mock Successful Mqtt Validation
+        when(mqttClient.createMqttConnectionBuilder(any(), any(), eq(null))).thenReturn(awsIotMqttConnectionBuilder);
+        when(awsIotMqttConnectionBuilder.withClientId(any())).thenReturn(awsIotMqttConnectionBuilder);
+        when(awsIotMqttConnectionBuilder.build()).thenReturn(mqttClientConnection);
+        CompletableFuture<Boolean> mqttConnection = CompletableFuture.completedFuture(true);
+        when(mqttClientConnection.connect()).thenReturn(mqttConnection);
+
+        // Mock Successful Http Validation
+        Pair<SdkHttpClient, GreengrassV2DataClient> pair = new Pair<>(null, greengrassV2DataClient);
+        when(gscFactory.createClientFromConfig(any())).thenReturn(pair);
+        when(greengrassV2DataClient.listThingGroupsForCoreDevice((ListThingGroupsForCoreDeviceRequest) any()))
+                .thenReturn(null);
+
+        // WHEN
+        Topics t = kernel.findServiceTopic(FleetStatusService.FLEET_STATUS_SERVICE_TOPICS);
+        assertNotNull(t, "FSS Topics should not be null before merging");
+        deploymentConfigMerger.mergeInNewConfig(testDeployment(), deploymentConfig).get(60, TimeUnit.SECONDS);
+        t = kernel.findServiceTopic(FleetStatusService.FLEET_STATUS_SERVICE_TOPICS);
+        assertNotNull(t, "FSS Topics should not be null after merging");
+
+        // THEN
+        deviceConfiguration = kernel.getContext().get(DeviceConfiguration.class);
+        int actualMqttPort = Coerce.toInt(deviceConfiguration.getMQTTNamespace().find(MqttClient.MQTT_PORT_KEY));
+        assertEquals(expectedMqttPort, actualMqttPort);
+        int actualDataPlanePort = Coerce.toInt(deviceConfiguration.getGreengrassDataPlanePort());
+        assertEquals(expectedDataPlanePort, actualDataPlanePort);
+    }
+
+    @Test
+    void GIVEN_kernel_running_with_some_config_WHEN_mqtt_validation_fails_THEN_config_is_not_updated(ExtensionContext context)
```

Comment:
Can you add a case for config validation interrupted?

Reply:
Added interruption exception test here and added tests to Connectivity Validator to cover exceptions